src/main/java/Duke.java

```diff
@@ -1,10 +1,221 @@
+                // No fallthrough required
+                switch (command) {
+                // If user wants to mark a task as done
+                case MARK:
+                    markAsDone(line);
+                    break;
+                // If user wants to list all tasks
+                case LIST:
+                    listItems();
+                    break;
+                case ADD:
+                    // If user wants to add an item
+                    addItem(line);
+                    break;
+                case EXIT:
+                    return;
+                }
+            }
+        }
+    }
+    /**
+     * Marks a task as done
+     * Checks for out of bounds access and presence of numerical value
+     * @param line user input
+     */
+    public static void markAsDone(String line) {
+        int listNum = checkValidDone(line);
+        if(listNum == INVALID_DONE) {
+            printError(ERR_OUT_OF_BOUNDS);
+        } else {
+            listNum -= 1;
+            Tasks[listNum].setDone();
+            System.out.println(Tasks[listNum].getName() + " set to done. You're well smart innit?");
+        }
+        printLine();
+    }
+
+    /**
+     * Checks for presence of number in input AND for any illegal access of indexes
+     * @param line input from user
+     * @return index of item to mark as done if valid, -1 otherwise
+     */
+    public static int checkValidDone(String line) {
+        if (line.substring(5).matches("[0-9]+")) {
+            int listNum = Integer.parseInt(line.substring(5));
+            // Check for illegal access to out of bounds index
+            if (listNum > Task.totalTasks || listNum == 0) {
+                return INVALID_DONE;
+            }
+            return listNum;
+        }
+        return INVALID_DONE;
+    }
+    /**
+     * Lists all items that were added to the list
+     */
+    public static void listItems() {
+        System.out.println("Eez are the tings you added to the list");
+        for(int i = 0; i < Task.totalTasks; i++) {
+            printListItem(i+1, Tasks[i].getType(), Tasks[i].getStatusIcon(),
+                    Tasks[i].getName(), Tasks[i].getDate());
+        }
+        printLine();
+    }
+    /**
+     * Append new item to back of the list
+     * @param line raw input given by user
+     */
+    public static void addItem(String line) {
+        String prefix = line.split(" ")[0];
+        int itemType = getItemType(prefix);
+        // No fallthrough required
+        switch (itemType) {
+        case ADD_TODO:
+            addTodo(line);
+            break;
+        case ADD_DEADLINE:
+            addDeadline(line);
+            break;
+        case ADD_EVENT:
+            addEvent(line);
+            break;
+        case ERR_WRONG_FORMAT:
+            printError(ERR_WRONG_FORMAT);
+        }
+        printLine();
+    }
+    public static void addTodo(String line) {
+        int current = Task.totalTasks;
```

Comment:
Variable naming should be clear. In this case, perhaps totalNumberOfTask works better.